packages/shorebird_cli/lib/src/commands/patch/patcher.dart

```diff
@@ -139,8 +141,17 @@ https://docs.shorebird.dev/status#link-percentage-ios
       return [];
     }
 
-    // If the user already provided --build-name or --build-number, we don't
-    // want to override them.
+    // If the user provided --build-name or --build-number before the --, we
+    // don't want to override them.
+    if (argResults.options.containsAnyOf([
+      CommonArguments.buildNameArg.name,
+      CommonArguments.buildNumberArg.name,
+    ])) {
+      return [];
+    }
+
+    // If the user provided --build-name or --build-number after the --, we
+    // don't want to override them.
```

Comment:
Should we refactor the next line to use the constant?

Reply:
Good call, done